ilc/server/tailor/configs-injector.js

```diff
@@ -246,4 +250,11 @@ module.exports = class ConfigsInjector {
         nrCode = nrCode.replace(/<script.*?>(.*)<\/script\s*>/s, '$1');
         return this.#nrCustomClientJsWrapper.replace('%CONTENT%', nrCode);
     };
+
+    #wrapWithRemoveBeforeParsing = (content) =>
+        `<!-- TailorX: Remove before parsing START -->${content}<!-- TailorX: Remove before parsing END -->`;
+
+    #wrapTagsToRemove(content, regex) {
```

Comment:
I cannot get why is it needed? Newrelic that is put here can be removed right away, these tags should be set directly in the templates if they supposed to be removed on LDE

Reply:
I tried to add a general logic for removing parts from templates and implement LDE check on ILC side. Not sure how we can implement this directly in templates.

Reply:
you just modify master templates directly on the registry

Reply:
Do you mean adding <!-- TailorX: Remove before parsing START --> to the templates directly? but if we do this, we will remove NewRelic not only for LDE. Could you elaborate more on your solution?

Reply:
Supposedly this tags should be implemented in a way they are removed only for LDE.
If we add these tags automatically we rely on the regexp to identify newrelic script within the template, which is a bad in a couple ways:
1) we adding new reliance to newrelic, while it is better to keep away it from ILC core
2) mechanics is not very reusable then, for other possible tools except newrelic, like sentry, datadog, etc. and not only limited to monitoring
3) newrelic might change the script making this regexp outdates, and functionality very fragile

Reply:
Do I understand right your solution:
1. Update template directly in the repository by wrapping NewRelic scripts in some custom tag
2. Add logic on ILC that removes all parts from the template that are wrapped in a custom tag (etc. if request is in LDE mode)

If it is true, how correctly update templates (by running some migration script or update them manually)? Do I understand right that only **master** template should be updated?

Reply:
Yeap, solution sound correct. 
As for changing templates there is no automated migration, as for our case we have around 4-6 master templates, it shouldn't be a big effort to update them.